packages/common/src/store/pages/saved-page/reducer.ts

```diff
@@ -241,7 +245,23 @@ const actionsMap: ActionsMap<SavedPageState> = {
   [SET_SELECTED_CATEGORY](state, action) {
     return {
       ...state,
-      selectedCategory: action.category
+      ...calculateNewLibraryCategories({
+        currentTab: action.currentTab,
+        chosenCategory: action.category,
+        prevTracksCategory: state.tracksCategory
+      })
+    }
+  },
+  [INIT_TRACKS_CATEGORY_FROM_LOCAL_STORAGE](state, action) {
```

Comment:
Can we use redux persist for all this?

Reply:
for posterity: we're going to do this in a separate PR

Reply:
https://linear.app/audius/issue/PAY-1816/use-redux-persist-for-persisting-selected-library-categories